src/components/Courses/Courses.tsx

```diff
@@ -1,62 +1,86 @@
 import { useNavigate } from 'react-router-dom';
+import { useEffect, useMemo, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '../../store/store';
+import { fetchCourses } from '../../store/courses/coursesSlice';
+import { fetchAuthors } from '../../store/authors/authorsSlice';
+
 import { CourseCard } from './components/CourseCard/CourseCard';
 import { getCourseDuration, formatCreationDate } from '../../helpers';
-import { ICourse, IAuthor } from '../../mockData';
 import { SearchBar } from './components/SearchBar/SearchBar';
 import { Button } from '../../common/Button/Button';
 import styles from './Courses.module.css';
 
-interface ICoursesProps {
-	courses: ICourse[];
-	authors: IAuthor[];
-	handleSearchInput: any;
-	handleClickSearchButton: any;
-	search: string;
-}
-
-export const Courses = ({
-	courses,
-	authors,
-	handleSearchInput,
-	handleClickSearchButton,
-	search,
-}: ICoursesProps) => {
+export const Courses = () => {
 	const navigate = useNavigate();
+	const dispatch = useAppDispatch();
+
+	const [showSearchCourses, setShowSearchCourses] = useState(false);
+
+	const coursesLoading = useAppSelector(
+		(state) => state.coursesReducer.coursesStatus
+	);
+
+	const authorsLoading = useAppSelector(
+		(state) => state.authorsReducer.authorsStatus
+	);
+
+	useEffect(() => {
+		dispatch(fetchCourses());
+		dispatch(fetchAuthors());
+	}, [dispatch]);
```

Comment:
not sure about passing dispatch to useEffect args, I'd keep them empty as that makes it immediately clear to the reader that this is meant to run on first render only

Reply:
We have warning without it

![image](https://user-images.githubusercontent.com/65336513/178771000-87968289-af24-453d-a3d6-22d05631b657.png)